src/main.cpp

```diff
@@ -106,11 +106,20 @@ int main(int argc, char** argv) {
 
     if (cli->has("config")) {
       confpath = cli->get("config");
-    } else if (env_util::has("XDG_CONFIG_HOME")) {
+    }
+    if (env_util::has("XDG_CONFIG_HOME") && confpath.empty()) {
       confpath = env_util::get("XDG_CONFIG_HOME") + "/polybar/config";
-    } else if (env_util::has("HOME")) {
+      if (!file_util::exists(confpath)) {
+        confpath = "";
+      }
+    }
+    if (env_util::has("HOME") && confpath.empty()) {
       confpath = env_util::get("HOME") + "/.config/polybar/config";
-    } else {
+      if (!file_util::exists(confpath)) {
+        confpath = "";
+      }
+    }
```

Comment:
I would suggest putting this part into a separate function here in `main.cpp`. This way you don't have to use the workaround of setting `confpath` to the empty string everytime and can just return early.

Reply:
I've made this change, but I'm not sure if the place that I putted the function is appropriated, I was thinking about put it on config class, but it would force polybar instantiate the config class earlier making this small change affect more the code. What do u think?

Reply:
I think it would also have been fine in `main.cpp`, but this is also good.